client/src/components/TabPane/Cards/MessagesCard.js

```diff
@@ -1,7 +1,462 @@
+          resetAdding();
+        } else {
+          error(res.message);
+        }
+      })
+      .catch((err) => {
+        console.log("Error while adding new message: ", err);
+        error("Error! Couldn't add message");
+      });
+  };
+
+  const fetchLanguages = () => {
+    fetch(`${baseURL}/language`)
+      .then((response) => response.json())
+      .then((data) => {
+        for (let i = 0; i < data.length; i++) {
+          data[i]['label'] = data[i]['language'];
+          data[i]['value'] = data[i]['language'];
+        }
+        if (data.length === 0)
+        {
+          data.push({'label': 'English', 'value': 'English'});
+        }
+        setLangData(data);
+      })
+      .catch((error) => {
+        console.error("Error while fetching langs: ", error);
+      });
+  };
+
+  const getOriginalMessages = (data) => {
+    let originals = []
+    for (let i = 0; i < data.length; i++) {
+      if (data[i]['original_message'] == null)
+      {
+        let newData = { 'label': data[i]['content'], 'value': data[i]['id']};
+        originals.push(newData);
+      }
+    }
+    originals.push({ 'label': 'None', 'value': null});
+    setoriginalMessData(originals);
+  }
+
+  const fetchTags = () => {
+    fetch(`${baseURL}/tag`)
+      .then((response) => response.json())
+      .then((data) => {
+        for (let i = 0; i < data.length; i++) {
+          data[i]['label'] = data[i]['tag'];
+          data[i]['value'] = data[i]['tag'];
+        }
+        setTagData(data);
+      })
+      .catch((error) => {
+        console.error("Error while fetching tags: ", error);
+      });
+  }
+
+  return (
+    <>
+      {resultMessageHolder}
+      <Row justify="end">
+        <Button
+          onClick={onAdd}
+          type="primary"
+          icon={<PlusOutlined />}
+          style={{ marginBottom: 16 }}
+        >
+          Add a new message
+        </Button>
+      </Row>
+
+      <Table
+        searchable
+        searchableProps={{
+          fuzzySearch: true,
+          fuzzyProps: {
+            ignoreLocation: true,
+            includeScore: true,
+            threshold: 0.0,
+          },
+        }}
+        columns={columns}
+        dataSource={data}
+        rowKey={(record) => record.id}
+        pagination={tableParams.pagination}
+        loading={loading}
+        size="middle"
+        onChange={handleTableChange}
+      />
+
+      <Modal
+        title="Edit message"
+        open={isEditing}
+        okText="Save"
+        onCancel={() => {
+          resetEditing();
+        }}
+        onOk={saveEdited}
+      >
+        <Select
+          style={{
+            width: '100%',
+          }}
+          value={editingMessage?.language}
+          placeholder="Please add language"
+          onChange={(e) => {
+            setEditingMessage((pre) => {
+              return { ...pre, language: e };
+            });
+          }}
+          options={langData}
+        />
+        <br/>
+        <Select
```

Comment:
Maybe in original message input add None as a default initial selected

Reply:
or better one, for English it should be None and disable and for any other we should be able to choose original message and the there shouldn't be available option None

Reply:
Do we really need this? I would prefer to leave it as it and rely on a backend responses. Adding conditional rendering might make this even more complex and on top of my head I cannot find any simpler and safer solution.

Reply:
It would be more intuitive, but if every scenario will be handled in the backend and we will get a proper response to the frontend, ofc it is not necessary

Reply:
I've added a change to disable `original_message` field for English messages.